### PR TITLE
test(e2e): Add e2e tests for internal reviews

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

This PR aims to add e2e tests for the new internal review functionality.

## Motivation and Context

There was no testing for the internal review functionality.

## How Has This Been Tested

- e2e tests
- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-3384

## Changes

Changes are mostly e2e tests related

## Depends on

- https://github.com/UserOfficeProject/user-office-core/pull/258
- https://github.com/UserOfficeProject/user-office-core/pull/259

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/user-office-frontend-e2e/cypress/e2e/internalReviews.cy.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+const proposal1 = {
+  title: faker.random.words(2),
+  abstract: faker.random.words(5),
+};
+let createdInstrumentId: number;
+let createdProposalPk: number;
+let technicalReviewId: number;
+let numberOfScientistsAndManagerAssignedToCreatedInstrument: number;
+
+context('Institution tests', () => {
```
Comment (Scott-James-Hurley) [inline apps/user-office-frontend-e2e/cypress/e2e/internalReviews.cy.ts:24]: Would a more appropriate name be 'Internal Review tests'?

File: apps/user-office-frontend-e2e/cypress/e2e/internalReviews.cy.ts [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+    cy.get('[data-cy="profile-page-btn"]').should('exist');
+
+    cy.get('[data-cy="user-menu-items"]').as('userMenuItems');
+
+    cy.get('@userMenuItems').should('not.contain', 'Review Proposals');
+    cy.get('@userMenuItems').should('not.contain', 'Review Proposals');
+  });
+
+  it('User Officer should be able to create and update internal review', () => {
```
Comment (Scott-James-Hurley) [inline apps/user-office-frontend-e2e/cypress/e2e/internalReviews.cy.ts:109]: Should creating and updating an internal review be separate tests?

File: apps/user-office-backend/src/datasources/mockups/InternalReviewDataSource.ts [hunk 3]
```diff
@@ -106,4 +106,18 @@ export class InternalReviewDataSourceMock implements InternalReviewDataSource {
...
+
+  async isInternalReviewerOnTechnicalReview(
+    userId: number,
+    technicalReviewId: number
+  ) {
+    return true;
+  }
+  async isInternalReviewer(userId: number) {
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (martin-trajanovski) [reply apps/user-office-backend/src/datasources/mockups/InternalReviewDataSource.ts]: @Junjiequan this is fixed already in one of the previous PRs. Now you can check only the e2e tests.

**Review by Scott-James-Hurley — APPROVED** [review]

File: apps/user-office-frontend-e2e/cypress/e2e/internalReviews.cy.ts [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+
+    cy.get('[data-cy="create-modal"]')
+      .find('[data-cy="title"] input')
+      .should('not.exist');
+
+    cy.get('[data-cy="create-modal"]')
+      .find('[data-cy="title"]')
+      .should('have.text', title);
+
+    // NOTE: For some reason one click on the update button is not enough. Probably because of tinymce content setting.
```
[the remark this answers is not in the document]
Comment (martin-trajanovski) [reply apps/user-office-frontend-e2e/cypress/e2e/internalReviews.cy.ts:299]: I tried to reproduce it manually but it doesn't seem to be the problem for the users. It seems that only cypress is having some issues and that's why I put the note to clarify why it is executed twice.

**Review by mutambaraf — APPROVED** [review]

**Pull request merged**